odl/solvers/functional/default_functionals.py

```diff
@@ -2193,6 +2198,169 @@ def __repr__(self):
                                          self.__norm.pwisenorm.exponent)
 
 
+class IndicatorSimplex(Functional):
+
+    """The indicator functional of a simplex in any dimension.
+
+    Notes
+    -----
+    The functional :math:`F` is given by:
+
+    .. math::
+        F(x)
+        =
+        \\begin{cases}
+            0 & \\text{if } x_i \geq 0 \\forall i \\text{ and } \sum_i x_i = r
+            \\\\
+            +\\infty & \\text{else.}
+        \\end{cases}
+
+    where `r` is the radius.
+    """
+
+    def __init__(self, space, radius=1):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `DiscreteLp` or `FnBase`
+            Domain of the functional.
+        radius : positive float, optional
+            Radius `r` of simplex.
+        """
+        super().__init__(space=space, linear=False, grad_lipschitz=np.nan)
+        self.radius = radius
+
+    def __call__(self, x):
+        """Return whether x is in the unit simplex."""
```

Comment:
You overloaded the wrong method, use `_call` instead.
The docstring is also a bit misleading since I would assume that it returns a boolean and not 0 or infinity. Actually, """Return \`\`self(x)\`\`.""" is enough here.